docs/docs/setup/rust-setup.md

```diff
@@ -168,3 +168,122 @@ CONFIGURATION=release cargo run --bin android # Release mode
 Running your project in release mode significantly enhances performance compared to debug mode. This is because the Rust compiler applies optimizations that improve runtime speed and reduce binary size, making your application more efficient.
 
 :::
+
+## Setup Halo2-Based rust project
+
+Similar to the [Setup Circom-based Rust project](#setup-circom-based-rust-project), start by running the following commands in your terminal:
+
+```sh
+mkdir mopro-example-app
+cd mopro-example-app
+cargo init --lib
+```
+
+This will create a new Rust project in the current directory. Next, add the required dependencies to the project. Edit your `Cargo.toml` to match the following:
+
+```toml
+[package]
+name = "mopro-example-app"
+version = "0.1.0"
+edition = "2021"
+
+# We're going to build a static library named mopro_bindings
+# This library name should not be changed
+[lib]
+crate-type = ["lib", "cdylib", "staticlib"]
+name = "mopro_bindings"
+
+# Adapters for different proof systems
+[features]
+default = ["mopro-ffi/halo2"]
+
+[dependencies]
+mopro-ffi = { git = "https://github.com/zkmopro/mopro.git", branch = "main" }
+rust-witness = "0.1.1"
```

Comment:
actually we don't need `rust-witness` for halo2 circuits.
(We should update cli template as well)